engines/griffon/engine.cpp

```diff
@@ -0,0 +1,335 @@
+			_floatText[i].framesLeft -= spd;
+			_floatText[i].y -= spd;
+			if (_floatText[i].framesLeft < 0)
+				_floatText[i].framesLeft = 0;
+		}
+
+		if (_floatIcon[i].framesLeft > 0) {
+			float spd = 0.5 * _fpsr;
+			_floatIcon[i].framesLeft -= spd;
+			_floatIcon[i].y -= spd;
+			if (_floatIcon[i].framesLeft < 0)
+				_floatIcon[i].framesLeft = 0;
+		}
+	}
+
+	if (_player.level == _player.maxLevel)
+		_player.exp = 0;
+
+	if (_player.exp >= _player.nextLevel) {
+		_player.level = _player.level + 1;
+		addFloatText("LEVEL UP!", _player.px + 16 - 36, _player.py + 16, 3);
+		_player.exp = _player.exp - _player.nextLevel;
+		_player.nextLevel = _player.nextLevel * 3 / 2; // 1.5
+		_player.maxHp = _player.maxHp + _player.level * 3;
+		if (_player.maxHp > 999)
+			_player.maxHp = 999;
+		_player.hp = _player.maxHp;
+
+		_player.swordDamage = _player.level * 14 / 10;
+		_player.spellDamage = _player.level * 13 / 10;
+
+		if (config.effects) {
+			int snd = playSound(_sfx[kSndPowerUp]);
+			setChannelVolume(snd, config.effectsVol);
+		}
+	}
+
+	_clipBg->copyRectToSurface(_clipBg2->getPixels(), _clipBg2->pitch, 0, 0, _clipBg2->w, _clipBg2->h);
+
+	Common::Rect rc;
+
+	rc.left = _player.px - 2;
+	rc.top = _player.py - 2;
+	rc.setWidth(5);
+	rc.setHeight(5);
+
+	_clipBg->fillRect(rc, 1000);
+
+	if (!_forcePause) {
+		for (int i = 0; i < 5; i++) {
+			if (_player.foundSpell[i])
+				_player.spellCharge[i] += 1 * _player.level * 0.01 * _fpsr;
+			if (_player.spellCharge[i] > 100)
+				_player.spellCharge[i] = 100;
+		}
+
+		if (_player.foundSpell[0]) {
+			_player.spellStrength += 3 * _player.level * .01 * _fpsr;
+		}
+
+		_player.attackStrength += (30 + 3 * (float)_player.level) / 50 * _fpsr;
+	}
+
+	if (_player.attackStrength > 100)
+		_player.attackStrength = 100;
+
+	if (_player.spellStrength > 100)
+		_player.spellStrength = 100;
+
+	_itemyloc += 0.75 * _fpsr;
+	while (_itemyloc >= 16)
+		_itemyloc -= 16;
+
+	if (_player.hp <= 0) {
+		theEnd();
+
+		_gameEnd = true;
+
+		return;
+	}
+
+	if (_roomLock) {
+		_roomLock = false;
+		for (int i = 1; i <= _lastNpc; i++)
+			if (_npcInfo[i].hp > 0)
+				_roomLock = true;
+	}
+
+	_cloudAngle += 0.1 * _fpsr;
+	while (_cloudAngle >= 360)
+		_cloudAngle = _cloudAngle - 360;
+
+	_player.hpflash = _player.hpflash + 0.1 * _fpsr;
+	if (_player.hpflash >= 2) {
+		_player.hpflash = 0;
+		_player.hpflashb = _player.hpflashb + 1;
+		if (_player.hpflashb == 2)
+			_player.hpflashb = 0;
+		if (config.effects && _player.hpflashb == 0 && _player.hp < _player.maxHp / 4) {
+			int snd = playSound(_sfx[kSndBeep]);
+			setChannelVolume(snd, config.effectsVol);
+		}
+	}
+
+	// _cloudsOn = false
+
+	if (_itemSelOn)
+		_player.itemselshade = _player.itemselshade + 2 * _fpsr;
+	if (_player.itemselshade > 24)
+		_player.itemselshade = 24;
+
+	for (int i = 0; i <= 4; i++)
+		if (_player.inventory[i] > 9)
+			_player.inventory[i] = 9;
+}
+
+void GriffonEngine::newGame() {
+	intro();
+
+	if (_shouldQuit)
+		return;
+
+	_player.px = 0;
```

Comment:
I personally think it would be cleaner to have a separate "reset" method on your player class that sets all these, rather than just having them all here in the engine.cpp file

Reply:
Good point. Fixed.